lib/quicly.c

```diff
@@ -1752,6 +1821,10 @@ static int prepare_acked_packet(quicly_conn_t *conn, struct st_quicly_send_conte
 
     if ((ret = prepare_packet(conn, s, min_space)) != 0)
         return ret;
+    /* FIXME move the window size check into prepare_packet to avoid sending an empty QUIC packet */
+    /* TODO return the remaining window that the sender can use */
+    if (s->send_window < (s->dst - s->target.packet->data.base) + min_space)
```

Comment:
What's min_space used for?

Reply:
As discussed in https://github.com/h2o/quicly/pull/13/files/797eafb9bc4aecbc6ade78367c517e97d238af55#r196601841, we should not allocate a packet buffer when a frame cannot be written to that frame.

`min_space` is the minimum space that is required to write a frame.